crates/p2p/src/commands.rs

```diff
@@ -1,17 +1,26 @@
 //! Commands for P2P implementation from operator implementation.
 
 use bitcoin::{hashes::sha256, OutPoint, XOnlyPublicKey};
+use libp2p::identity::secp256k1;
 use musig2::{PartialSignature, PubNonce};
 use prost::Message;
+use strata_p2p_types::OperatorPubKey;
 use strata_p2p_wire::p2p::v1::{
-    DepositNonces, DepositSetup, DepositSigs, GenesisInfo, GossipsubMsgDepositKind,
+    DepositNonces, DepositSetup, DepositSigs, GenesisInfo, GossipsubMsg, GossipsubMsgDepositKind,
     GossipsubMsgKind,
 };
 
 /// Ask P2P implementation to distribute some data across network.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[allow(clippy::enum_variant_names)] /* remove this later, when other commands will be added */
-pub(crate) enum Command<DepositSetupPayload> {
+pub struct Command<DepositSetupPayload> {
+    pub key: OperatorPubKey,
+    pub signature: Vec<u8>,
+    pub kind: CommandKind<DepositSetupPayload>,
+}
```

Comment:
I don't think that `signature` and `key` should be here. For example, assume that `CommandKind` lacks the `RequestMessage` variant, which doesn't require `signature` and `key` fields, so, in this case, you couldn't just simply add new variant to `CommandKind` you would need to remake `Command` completly